lib/dynflow/tasks/export_task.rb

```diff
@@ -65,32 +70,50 @@ def task_ids
         @task_ids = ENV['TASK_IDS'].split(',') if ENV['TASK_IDS']
       end
 
+      def task_compress
+        ENV['TASK_COMPRESS'] || @task_compress || 'yes'
+      end
+
       private
 
+      def compress?
+        task_compress.downcase == 'yes'
+      end
+
       def export
-        if plans.empty?
+        if plans_or_ids.empty?
           puts("Nothing to export, exiting")
           return
         end
 
-        puts "Exporting #{plans.count} tasks"
-        content = case task_format
-                  when 'html'
-                    ::Dynflow::Exporters::Tar.full_html_export plans
-                  when 'json'
-                    ::Dynflow::Exporters::Tar.full_json_export plans
-                  when 'csv'
-                    ::Dynflow::Exporters::CSV.new
-                      .add_many(plans).result
-                  else
-                    raise "Unknown export format '#{format}'"
-                  end
-        File.write(task_file, content)
+        puts "Exporting #{plans_or_ids.count} tasks"
+        File.open(task_file, 'w') do |f|
+          f = Zlib::GzipWriter.new(f) if compress?
```

Comment:
refactoring suggestiong: extract the potential comprssion into `with_comprission(f, block&)` that would look something like:

```ruby
def with_compression(f, &block)
  if compress?
    begin
      gzip_f = Zlib::GzipWriter.new(f)
      yield gzip_f
    ensure
      gzip_f.close
    end
  else
    yield f
  end
end
```